### ISSUE TITLE
Log full of "Trigger not set"

### ISSUE THREAD

**Issue by julimej** [jenkinsci/bitbucket-push-and-pull-request-plugin#248]
Version:
Jenkins 2.327
Bitbucket PPR 2.8.1

When the plugin detects a new Push ir prints multiple lines of:
```
Trigger not set
Feb 20, 2022 4:20:47 PM WARNING io.jenkins.plugins.bitbucketpushandpullrequest.BitBucketPPRJobProbe lambda$triggerMatchingJobs$1

```
I think this happens on a while, scanning i dont know what. 

It can be produced by any job that is triggered by a Push and with the webhook configured.

**Comment by savoiringfaire** [comment]
Seeing this also - top of the logs once an MR has been opened and seeing:

```
2022-03-01 18:33:18.380+0000 [id=91159914]      INFO    i.j.p.b.r.BitBucketPPRHookReceiver#doIndex: Received POST request over Bitbucket hook
2022-03-01 18:33:18.384+0000 [id=91159914]      INFO    i.j.p.b.p.BitBucketPPRPayloadProcessorFactory#createProcessor: Create BitBucketPPRPullRequestServerPayloadProcessor
2022-03-01 18:33:18.396+0000 [id=91159914]      WARNING i.j.p.b.BitBucketPPRJobProbe#lambda$triggerMatchingJobs$1: Trigger not set
```

That last line is then repeated 6570 times within the next half a second.

with Jenkins version 2.289 and plugin version 2.8.1.

**Comment by xJom** [comment]
Same here. I also get an empty payload. Bitbucket says it sends the payload, but when the plugin try to interpret the payload it renders an empty set.

**Comment by klin111** [comment]
are you    fixed ?

**Comment by y0zg** [comment]
Received this error as well. Any updates?

**Comment by y0zg** [comment]
@cdelmonte-zg Do you know any details on this issue?

**Comment by y0zg** [comment]
Fixed for myself.  For me the issue was that Jenkinsfile wasn't present

**Comment by ofri-7chairs** [comment]
same here...
Plugin version: 2.8.1
Jenkins 2.375.1

`Jan 05, 2023 10:14:13 AM INFO io.jenkins.plugins.bitbucketpushandpullrequest.receiver.BitBucketPPRHookReceiver doIndex
Received POST request over Bitbucket hook
Jan 05, 2023 10:14:13 AM FINEST io.jenkins.plugins.bitbucketpushandpullrequest.receiver.BitBucketPPRHookReceiver
the payload is: {}
Jan 05, 2023 10:14:13 AM INFO io.jenkins.plugins.bitbucketpushandpullrequest.processor.BitBucketPPRPayloadProcessorFactory createProcessor
Create BitBucketPPRPullRequestCloudPayloadProcessor
Jan 05, 2023 10:14:13 AM FINE io.jenkins.plugins.bitbucketpushandpullrequest.observer.BitBucketPPRObserverFactory
Add BitBucketPPRPullRequestCloudObserver for {}
Jan 05, 2023 10:14:13 AM WARNING io.jenkins.plugins.bitbucketpushandpullrequest.BitBucketPPRJobProbe lambda$triggerMatchingJobs$1
Trigger not set
Jan 05, 2023 10:14:13 AM WARNING io.jenkins.plugins.bitbucketpushandpullrequest.BitBucketPPRJobProbe lambda$triggerMatchingJobs$1
Trigger not set
Jan 05, 2023 10:14:13 AM WARNING io.jenkins.plugins.bitbucketpushandpullrequest.BitBucketPPRJobProbe lambda$triggerMatchingJobs$1
Trigger not set
Jan 05, 2023 10:14:13 AM WARNING io.jenkins.plugins.bitbucketpushandpullrequest.BitBucketPPRJobProbe lambda$triggerMatchingJobs$1
Trigger not set
Jan 05, 2023 10:14:13 AM WARNING io.jenkins.plugins.bitbucketpushandpullrequest.BitBucketPPRJobProbe lambda$triggerMatchingJobs$1
Trigger not set
Jan 05, 2023 10:14:13 AM WARNING io.jenkins.plugins.bitbucketpushandpullrequest.BitBucketPPRJobProbe lambda$triggerMatchingJobs$1
Trigger not set`

**Comment by philippeqc** [comment]
Using the following, I observed the same issue with multiple "Trigger not set" in the log.
- Jenkins: 2.387.1
- Bitbucket Push and Pull Request Plugin: 2.8.3

I noticed that I had exactly one less "Trigger not set" than the number of jobs on the server. Also, I only had 1 job that was listening to those triggers. So there was some interaction between the plugin and my job, just no message ending in the log about it.

That job was set with `agent none` and no stage had any agent defined. Defining a valid agent made it work immediatly for me.

```
pipeline {
    agent {
        label 'something'
    }
    ...
```

Hopefully this will be of use to address the issue.
